source/shadow-tasks/shadow_update_task.c

```diff
@@ -0,0 +1,631 @@
+
+/**
+ * @brief Passed into MQTTAgent_Subscribe() as the callback to execute when the
+ * broker ACKs the SUBSCRIBE message.  Its implementation sends a notification
+ * to the task that called MQTTAgent_Subscribe() to let the task know the
+ * SUBSCRIBE operation completed.  It also sets the xReturnStatus of the
+ * structure passed in as the command's context to the value of the
+ * xReturnStatus parameter - which enables the task to check the status of the
+ * operation.
+ *
+ * See https://freertos.org/mqtt/mqtt-agent-demo.html#example_mqtt_api_call
+ *
+ * @param[in] pxCommandContext Context of the initial command.
+ * @param[in] pxReturnInfo The result of the command.
+ */
+static void prvSubscribeCommandCallback( void * pxCommandContext,
+                                         MQTTAgentReturnInfo_t * pxReturnInfo );
+
+/**
+ * @brief The callback to execute when there is an incoming publish on the
+ * topic for accepted requests. It verifies the document is valid and is being waited on.
+ * If so it notifies the task to inform completion of the update request.
+ *
+ * @param[in] pvIncomingPublishCallbackContext Context of the initial command.
+ * @param[in] pxPublishInfo Deserialized publish.
+ */
+static void prvIncomingPublishUpdateAcceptedCallback( void * pxSubscriptionContext,
+                                                      MQTTPublishInfo_t * pxPublishInfo );
+
+
+
+/**
+ * @brief The callback to execute when there is an incoming publish on the
+ * topic for rejected requests. It verifies the document is valid and is being waited on.
+ * If so it notifies the task to inform completion of the update request.
+ *
+ * @param[in] pvIncomingPublishCallbackContext Context of the initial command.
+ * @param[in] pxPublishInfo Deserialized publish.
+ */
+static void prvIncomingPublishUpdateRejectedCallback( void * pxSubscriptionContext,
+                                                      MQTTPublishInfo_t * pxPublishInfo );
+
+/**
+ * @brief Entry point of shadow demo.
+ *
+ * This main function demonstrates how to use the macros provided by the
+ * Device Shadow library to assemble strings for the MQTT topics defined
+ * by AWS IoT Device Shadow. It uses these macros for topics to subscribe
+ * to:
+ * - SHADOW_TOPIC_STRING_UPDATE_DELTA for "$aws/things/thingName/shadow/update/delta"
+ * - SHADOW_TOPIC_STRING_UPDATE_ACCEPTED for "$aws/things/thingName/shadow/update/accepted"
+ * - SHADOW_TOPIC_STRING_UPDATE_REJECTED for "$aws/things/thingName/shadow/update/rejected"
+ *
+ * It also uses these macros for topics to publish to:
+ * - SHADOW_TOPIC_STIRNG_DELETE for "$aws/things/thingName/shadow/delete"
+ * - SHADOW_TOPIC_STRING_UPDATE for "$aws/things/thingName/shadow/update"
+ */
+void vShadowUpdateTask( void * pvParameters );
+
+/*-----------------------------------------------------------*/
+
+static bool prvSubscribeToShadowUpdateTopics( void )
+{
+    bool xReturnStatus = true;
```

Comment:
maintainability: return status values should be initialized to false, e.g. `bool xReturnStatus = false;`  Note that you'll need to make adjustments in lines 290-299 for this.